app/controllers/users/sessions_controller.rb

```diff
@@ -206,6 +207,32 @@ def sign_in_params
     end
   end
 
+  def check_password_compromised
+    return if current_user.check_password_compromised_at.present? && 
+      eligible_for_password_lookup?
+
+    user_session[:redirect_to_phone_compromised] = PwnedPasswords::LookupPassword.call(auth_params[:password])
+    update_user_check_password_compromised_at
+  end
+
+
+  def eligible_for_password_lookup?
+    # We want to have a threshold to see how many users fall into this to do a check
+    IdentityConfig.store.check_user_password_compromised_enabled && 
+      randomize_check_password?
+  end
+
+  def update_user_check_password_compromised_at
+    UpdateUser.new(
+      user: current_user,
+      attributes: { check_password_compromised_at: Time.zone.now },
+    ).call
+  end
+
+  def randomize_check_password?
+    SecureRandom.random_number(1000) > 900 
```

Comment:
Did we consider using an A/B test for this? Or a config value for the check threshold?

Reply:
yup, I updated that later, I was just doing this as a draft while I got it running